### PR TITLE
fix: double RUST_MIN_STACK to resolve stack overflow

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

when running SQL planner test, we meet a stack overflow panic https://buildkite.com/risingwavelabs/pull-request/builds/63818#0193962f-47f4-484f-bd3e-21aa7aa25ae5
and same observation by @Li0k 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Li0k — APPROVED** [review]
LGTM

**Comment by Li0k** [comment]
qq: Can we also change the risedev test command with RUST_MIN_STACK ?

File: Makefile.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ env_scripts = [
   set_env RW_TELEMETRY_TYPE "test"
   set_env RW_SECRET_STORE_PRIVATE_KEY_HEX "0123456789abcdef0123456789abcdef"
   set_env RW_TEMP_SECRET_FILE_DIR "${PREFIX_SECRET}"
+  set_env RUST_MIN_STACK 8388608
```
Comment (tabVersion) [inline Makefile.toml:32]: @Li0k `risedev` command utilizes the Makefile here. So it has been applied to `risedev test`

**Review by BugenZhao — COMMENTED** [review]
I think we'd better investigate which phase of planning requires the most stack space (and whether or why it grows recently), then adopt https://github.com/risingwavelabs/risingwave/pull/16279 to enable stack growing there.

File: Makefile.toml (hunk 1 above)
Comment (BugenZhao) [reply Makefile.toml:32]: I suggest not touching this, or at least only set if it's unset.

Comment (tabVersion) [reply Makefile.toml:32]: But running `risedev test` locally also fails. I am afraid we cannot accept this.

Comment (tabVersion) [reply Makefile.toml:32]: Yes, I am checking the PR you mentioned.

**Comment by tabVersion** [comment]
Setting RUST_MIN_STACK to 8388608 (8MB) just when running `risedev test` and UT section in CI. Leave RUST_MIN_STACK to 4194304 ib other cases.

File: Makefile.toml [hunk 2]
```diff
@@ -1331,7 +1340,7 @@ SLT_HOST = "${RISEDEV_RW_FRONTEND_LISTEN_ADDRESS}"
 SLT_PORT = "${RISEDEV_RW_FRONTEND_PORT}"
 SLT_DB = "dev"
 PATH = "${PWD}/e2e_test/commands:${PATH}"
-RUST_BACKTRACE = "0" # slt backtrace is useless
+RUST_BACKTRACE = "0"                                             # slt backtrace is useless
```
Comment (st1page) [inline Makefile.toml:1343]:
```suggestion
RUST_BACKTRACE = "0" # slt backtrace is useless
```

**Pull request merged**